### DISCUSSION TITLE
General availability of Microsoft.PowerFx.Dataverse.Eval

### DISCUSSION THREAD

**Discussion by Zerajima** [microsoft/Power-Fx#2659] [Q&A]
Looking at [Power Fx samples repository](https://github.com/microsoft/power-fx-host-samples) I have noticed a Dataverse101 sample app, which uses Microsoft.PowerFx.Dataverse.Eval library. It uses a pre-release version of this library from last October, and this is also the only version available publically on nuget.org. I checked the Power platform CLI tool and saw it also uses this library, but a newer version of it. And I'm guessing you also use it internally for low-code plugins.

Do you plan to release this library to the public? If so, do you know when we could expect it?

**Comment by MikeStall** [comment]
All correct! And this is indeed what we use for low-code plugins. 

We do plan to refresh the public nuget, it just hasn't been a priority.  Can you tell me more about how you're using it?  Did you have any requestrs/feedback on the nuget?

**Reply by Zerajima** [reply]
We are working on a document automation solution for Power Platform, and we are looking for a way to automate certain inputs with expressions, (i.e. file name of a generated document, recipient e-mail address, etc.). Naturally, we thought about Power Fx as a way to achieve this, and having access to Dataverse data would make it much more flexible, than us having to provide the engine with context data ahead of time.

As for the feedback, I have only just started looking into this, so I don't really have any right now. I have tried the sample to see how it works and so far it looks very promising, pretty much exactly what we need. It will probably take another month or two, before we actually start working on integrating Power Fx into our solution. Is there any chance the package could be updated by the end of the year?

**Comment by jureles** [comment]
I'm wondering, is there any news on the release yet? Thanks!
